### PR TITLE
dotnet watch refactoring

### PR DESCRIPTION
**Pull request by tmat**
Simplifies implementation.

Removes unnecessary abstractions and mutable state.

### CODE REVIEW

**Comment by tmat** [comment]
@tlmii PTAL. Each commit can be reviewed separately.

File: src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs [hunk 1]
```diff
@@ -11,55 +11,38 @@
...
 
             async Task<ImmutableArray<string>> GetApplyUpdateCapabilitiesCoreAsync()
             {
-                if (context.BrowserRefreshServer is null)
-                {
-                    throw new ApplicationException("The browser refresh server is unavailable.");
-                }
-
-                _reporter.Verbose("Connecting to the browser.");
+                reporter.Verbose("Connecting to the browser.");
```
Comment (WardenGnaw) [inline src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs:36]: Do you need a browserRefreshServer null check here or if not, do a Debug.Assert check?
```
            if (browserRefreshServer is null)
            {
                throw new ApplicationException("The browser refresh server is unavailable.");
            }
```

File: src/BuiltInTools/dotnet-watch/dotnet-watch.csproj [hunk 2]
```diff
@@ -43,39 +43,17 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\BrowserRefresh\Microsoft.AspNetCore.Watch.BrowserRefresh.csproj"
```
Comment (WardenGnaw) [inline src/BuiltInTools/dotnet-watch/dotnet-watch.csproj:46]: Was the file formatting done on purpose? One attribute per line makes csprojs easier to read.

**Review by WardenGnaw — APPROVED** [review]

File: src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs (hunk 1 above)
Comment (tmat) [reply src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs:36]: No need for neither. `BrowserRefreshServer browserRefreshServer` parameter is declared as not-null (nullability checks are enabled in the entire project).

File: src/BuiltInTools/dotnet-watch/dotnet-watch.csproj (hunk 2 above)
Comment (tmat) [reply src/BuiltInTools/dotnet-watch/dotnet-watch.csproj:46]: Not on purpose, VS did it automatically :( Will undo.

**Review by phil-allen-msft — APPROVED** [review]

**Pull request merged**